### ISSUE TITLE
Composer update throws a "The following untracked working tree files would be overwritten by"

### ISSUE THREAD

**Issue by mwjames** [SemanticMediaWiki/SemanticMediaWiki#1351]
https://www.semantic-mediawiki.org/wiki/Thread:Help_talk:Using_Composer/update_composer_fails

```
[RuntimeException]                                                                                                                              
  Failed to execute git checkout 'f7543bcb7ad1b08c1924d65aeea9e353d5953381' -- && git reset --hard 'f7543bcb7ad1b08c1924d65aeea9e353d5953381' --  
  error: The following untracked working tree files would be overwritten by checkout:                                                             
  	tests/phpunit/Integration/ByJsonScript/Fixtures/p-0408 [#1252] intext to use non-strict mode.json                                              
  Please move or remove them before you can switch branches.                                                                                      
  Aborting 
```

**Comment by mwjames** [comment]
@kghbln I think https://github.com/composer/composer/issues/2896 and https://github.com/composer/composer/pull/3789 are relevant. Did you use a recent Composer version (which would include the latter) ?

I assume that due to using a specific github-branch at an earlier occasion, the git tree was out of sync and some files were missing the correct hash signature. As indicated by https://github.com/composer/composer/pull/3789, option [discard-changes](https://getcomposer.org/doc/06-config.md#discard-changes) should help finish the update.

**Comment by kghbln** [comment]
Well, I did the brute force method I described on the wiki. I had changes to some files before, but was always asked if I would like to continue instead of getting an advice I took too serious. Probably I should indeed add the `discard-changes` option to my "composer.json" file.

Bevore I forget: I currently use `Composer version 1.0-dev (72cd6afdfce16f36a9fd786bc1b2f32b851e764f) 2015-12-28 17:35:19` on the respective server.
